app/main.py

```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached = {}
+
+    def inner(*args, **kwargs) -> Callable:
+        key = f"{func.__name__} {str(args)} {str(kwargs)}"
+        if cached.get(key) is None:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cached[key] = result
+            return result
```

Comment:
Think how you can call `return` only once . If  cached[key] = `result` you can return `cached[key]` and it is makes 2 identical `returns` in your func . Optimise it : )